pallets/gear-program/src/migrations/v13_split_instrumented_code_migration.rs

```diff
@@ -0,0 +1,275 @@
+use frame_support::{
+    traits::{Get, GetStorageVersion, OnRuntimeUpgrade, StorageVersion},
+    weights::Weight,
+};
+use gear_core::code::{CodeMetadata, InstrumentedCode};
+use sp_std::marker::PhantomData;
+
+use gear_core::code::InstrumentationStatus;
+#[cfg(feature = "try-runtime")]
+use {
+    frame_support::ensure,
+    sp_runtime::{
+        codec::{Decode, Encode},
+        TryRuntimeError,
+    },
+    sp_std::vec::Vec,
+};
+
+const MIGRATE_FROM_VERSION: u16 = 12;
+const MIGRATE_TO_VERSION: u16 = 13;
+const ALLOWED_CURRENT_STORAGE_VERSION: u16 = 13;
+
+pub struct MigrateSplitInstrumentedCode<T: Config>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for MigrateSplitInstrumentedCode<T> {
+    fn on_runtime_upgrade() -> Weight {
+        // 1 read for onchain storage version
+        let mut weight = T::DbWeight::get().reads(1);
+        let mut counter = 0;
+
+        let onchain = Pallet::<T>::on_chain_storage_version();
+
+        if onchain == MIGRATE_FROM_VERSION {
+            let current = Pallet::<T>::in_code_storage_version();
+
+            if current != ALLOWED_CURRENT_STORAGE_VERSION {
+                log::error!("❌ Migration is not allowed for current storage version {current:?}.");
+                return weight;
+            }
+
+            let update_to = StorageVersion::new(MIGRATE_TO_VERSION);
+
+            log::info!("🚚 Running migration from {onchain:?} to {update_to:?}, current storage version is {current:?}.");
+
+            v12::CodeStorage::<T>::drain().for_each(|(code_id, instrumented_code)| {
+                // 1 read for instrumented code, 1 write for instrumented code and 1 write for code metadata
+                weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 2));
+
+                let code_metadata = CodeMetadata::new(
+                    instrumented_code.original_code_len,
+                    instrumented_code.code.len() as u32,
+                    instrumented_code.exports,
+                    instrumented_code.static_pages,
+                    instrumented_code.stack_end,
+                    InstrumentationStatus::Instrumented(instrumented_code.version),
+                );
+
+                let instrumented_code = InstrumentedCode::new(
+                    instrumented_code.code,
+                    instrumented_code.instantiated_section_sizes,
+                );
+
+                InstrumentedCodeStorage::<T>::insert(code_id, instrumented_code);
+                CodeMetadataStorage::<T>::insert(code_id, code_metadata);
+
+                counter += 1;
+            });
+
+            v12::CodeStorageNonce::<T>::kill();
+            // killing a storage: one write
+            weight = weight.saturating_add(T::DbWeight::get().writes(1));
+
+            v12::CodeLenStorageNonce::<T>::kill();
+            // killing a storage: one write
+            weight = weight.saturating_add(T::DbWeight::get().writes(1));
```

Comment:
I do not see these nonces in `master`. The migration should be actualized

Reply:
removed non-existent nonce deletion